warehouse/query-core/src/main/java/datawave/query/tables/keyword/KeywordChainedUUIDQueryLogic.java

```diff
@@ -0,0 +1,101 @@
+package datawave.query.tables.keyword;
+
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.log4j.Logger;
+
+import datawave.core.query.configuration.GenericQueryConfiguration;
+import datawave.core.query.logic.QueryLogicTransformer;
+import datawave.microservice.query.Query;
+import datawave.microservice.query.QueryImpl;
+import datawave.query.QueryParameters;
+import datawave.query.tables.chained.ChainedQueryTable;
+
+public class KeywordChainedUUIDQueryLogic extends ChainedQueryTable<Entry<Key,Value>,Entry<Key,Value>> {
+
+    private static final Logger log = Logger.getLogger(KeywordChainedUUIDQueryLogic.class);
+
+    private Query keywordExtractionQuery = null;
+
+    public KeywordChainedUUIDQueryLogic() {
+        super();
+    }
+
+    public KeywordChainedUUIDQueryLogic(KeywordChainedUUIDQueryLogic other) {
+        super(other);
+    }
+
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    @Override
+    public GenericQueryConfiguration initialize(AccumuloClient client, Query settings, Set<Authorizations> auths) throws Exception {
+        super.initialize(client, settings, auths);
+        this.keywordExtractionQuery = settings.duplicate(settings.getQueryName() + "_keyword_query");
+
+        QueryImpl.Parameter hitListParam = settings.findParameter(QueryParameters.HIT_LIST);
+        if (hitListParam.getParameterValue().isEmpty()) {
+            settings.addParameter(QueryParameters.HIT_LIST, "true");
+        } else {
+            hitListParam.setParameterValue("true");
+        }
+
+        QueryImpl.Parameter returnFieldsParam = settings.findParameter(QueryParameters.RETURN_FIELDS);
+        if (returnFieldsParam.getParameterValue().isEmpty()) {
+            settings.addParameter(QueryParameters.RETURN_FIELDS, "LANGUAGE");
+        } else {
+            returnFieldsParam.setParameterValue("LANGUAGE");
+        }
+
+        QueryImpl.Parameter querySyntaxParam = settings.findParameter(QueryParameters.QUERY_SYNTAX);
+        if (querySyntaxParam.getParameterValue().isEmpty()) {
+            settings.addParameter(QueryParameters.QUERY_SYNTAX, "LUCENE-UUID");
+        } else {
+            querySyntaxParam.setParameterValue("LUCENE-UUID");
+        }
+
+        log.debug("Initial settings parameters: " + settings.getParameters().toString());
+        return this.logic1.initialize(client, settings, auths);
+    }
```

Comment:
maybe move the configuration checks from setupQuery() up here. Also add checks for a logic1 and logic2 being set?